src/IO/WriteBufferFromS3.cpp

```diff
@@ -126,6 +188,11 @@ WriteBufferFromS3::~WriteBufferFromS3()
     }
 }
 
+bool WriteBufferFromS3::cacheEnabled() const
+{
+    return cache != nullptr && !IFileCache::shouldBypassCache();
```

Comment:
suspicious, need to split this check in parts, it is incorrect now for background merges (functional test passes because there is FINAL)